server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java

```diff
@@ -151,4 +168,32 @@ private Mail toMail(EnqueuedItem enqueuedItem, MimeMessage mimeMessage) {
             .range(0, configuration.getBucketCount())
             .map(BucketId::of);
     }
+
+    public static class CassandraMailQueueItemView implements ManageableMailQueue.MailQueueItemView {
+        private final EnqueuedItem enqueuedItem;
+        private final Mail mail;
+
+        public CassandraMailQueueItemView(Pair<EnqueuedItem, Mail> pair) {
+            this(pair.getLeft(), pair.getRight());
+        }
+
+        public CassandraMailQueueItemView(EnqueuedItem enqueuedItem, Mail mail) {
+            this.enqueuedItem = enqueuedItem;
+            this.mail = mail;
+        }
+
+        public EnqueuedItem getEnqueuedItem() {
```

Comment:
I would say:

 - You likely need less than the enqueuedItem?
 - enqueuedItem also hold a Mail which is error prone as I can do `.getMail()` or `.getEnqueuedItem().getMail()` and might get a different result.

I would extract all the fields of EnqueuedItem that are not a mailn and add a getter for them here in order not to expose the EnqueuedItem.